grr/core/grr_response_core/lib/type_info.py

```diff
@@ -371,6 +370,21 @@ def ToString(self, value):
     return ",".join([self.validator.ToString(x) for x in value])
 
 
+class RDFValueList(List):
+  """A List of RDFValue."""
+  class _validator:
+    def __init__(self, validator=None):
+      self.validator = validator
+    def Validate(self, val):
+      try:
+        return self.validator.Validate(val.encode('utf-8'))
+      except Exception as e:
+        raise TypeValueError("Invalid data: %s" % e)
+  def __init__(self, rdfclass=None, validator=None, **kwargs):
+      super(RDFValueList, self).__init__(**kwargs)
+      self.validator = self._validator(validator)
+      self._type = self.rdfclass = rdfclass
+
```

Comment:
This will make an already complicate GRR type info system even more complicated :)
What I'd suggest instead is to implement an RDFX509CertChain class that would know how to parse from/serialize into a canonical representation of a cert chain. The call can be done similarly to the RDFX509Cert (https://github.com/google/grr/blob/c2a442a27f656fb18dfa3bce098847e5c5b849d7/grr/core/grr_response_core/lib/rdfvalues/crypto.py#L67). But instead of having cert-specific methods, it will have a "certs" property that will return a list of certs in the chain (each having RDFX509Cert type). 
Also, this class would have to be tested to ensure parsing works as expected.